### PR TITLE
Added log volume and logging for IQ 113

### PR DESCRIPTION
**Pull request by srini-hdp**
IQ 113 Docker Image have introduced start.sh script that stores stderr to /var/log/nexus-iq-server and adds the volume.
Added PVC / PV for log volume in Helm chart, to fix the below error during helm deployment.

`./start.sh: line 1: /var/log/nexus-iq-server/stderr.log: Permission denied`

Also enabled IQ loggers to use the log files in log volume

### CODE REVIEW

**Comment by srini-hdp** [comment]
Have used the IQ standard logging level: DEBUG as default option in helm as well to have same level of deployment as standalone instance.

File: charts/nexus-iq/templates/deployment.yaml [hunk 1]
```diff
@@ -56,6 +56,8 @@ spec:
           volumeMounts:
             - mountPath: /sonatype-work
               name: nxiq-pv-data
+            - mountPath: /var/log/nexus-iq-server
```
Comment (peterlynch) [inline charts/nexus-iq/templates/deployment.yaml:59]: Why /var/log/nexus-iq-server? Is /var/log some requirement of deployments?

From product perspective we like all log files to be under the same work dir, like the defaults of a normal on premise deployment.

File: charts/nexus-iq/values.yaml [hunk 2]
```diff
@@ -144,17 +147,20 @@ configYaml:
     requestLog: 
       appenders:
         #All appenders set to console
-        - type: console
```
Comment (jflinchbaugh) [inline charts/nexus-iq/values.yaml:147]: previously, our servers dumped all this logging to the console to be viewed as kubectl logs. Does directing the logs to files disable some monitoring that a k8s user would have expected?

File: charts/nexus-iq/templates/deployment.yaml (hunk 1 above)
Comment (jflinchbaugh) [reply charts/nexus-iq/templates/deployment.yaml:59]: /var/log/nexus-iq-server is the standard place referenced by the docker docs (https://hub.docker.com/r/sonatype/nexus-iq-server), so this seems consistent. in my testing, this new mount point here allows the logs to be persisted across restart. 👍

File: charts/nexus-iq/values.yaml (hunk 2 above)
Comment (jflinchbaugh) [reply charts/nexus-iq/values.yaml:147]: tested, and i see that i still see logs on console, so i guess that's good.

File: charts/nexus-iq/values.yaml [hunk 3]
```diff
@@ -144,17 +147,20 @@ configYaml:
...
+          archivedLogFilenamePattern: /var/log/nexus-iq-server/request-%d.log.gz
+          archivedFileCount: 50
   
   createSampleData: true
 
   logging:
 
     # The default level of all loggers. Can be OFF, ERROR, WARN, INFO, DEBUG, TRACE, or ALL.
-    level: ERROR
+    level: DEBUG
```
Comment (jflinchbaugh) [inline charts/nexus-iq/values.yaml:163]: i'm now seeing debug-level logs in clm-server.log. 👍

**Review by jflinchbaugh — APPROVED** [review]
+1

**Comment by jflinchbaugh** [comment]
@srini-hdp this tested fine with minikube. I tried with CRC (openshift) but ran into some permission errors. I found that may have been an existing problem on master with my environment, so if you like this for openshift, then that's good. i can merge this.

File: charts/nexus-iq/templates/deployment.yaml (hunk 1 above)
Comment (srini-hdp) [reply charts/nexus-iq/templates/deployment.yaml:59]: Yes I picked the directory from our docker docs

**Comment by srini-hdp** [comment]
I've done a sanity testing on OpenShift CRC in my Mac.

**Pull request merged**